src/redux/actions/loginAction.js

```diff
@@ -0,0 +1,46 @@
+import { LOGIN_REQUEST, LOGIN_SUCCESS, LOGIN_FAIL } from "../types/loginTypes";
+import axios from "axios";
+
+export const loginAction = (data,history) => async (dispatch) => {
+
+    try {
+    dispatch(loginRequest());
+    const res = await axios.post("https://eagles-phantom-backend.herokuapp.com/api/v1/auth/login", data);
+
+    const user = await res.data;
+    localStorage.setItem("token", user.token);
+    localStorage.setItem('user-data', JSON.stringify(user));
+
+    history.push('/dashboard')
+    dispatch(loginSuccess(user));
+
+    
+    } catch (error) {
+      if (error.response) {
+        const errorMessage = await error.response.data.message;
+        dispatch(loginFails(errorMessage))
+    }
+    else{
+        dispatch(loginFails("Error, please check your connection and try again!"))
+    }
+    }
+};
+
+export const loginRequest = () => {
+  return {
+    type: LOGIN_REQUEST,
+  };
+};
+
+export const loginSuccess = (data) => {
+  return {
+    type: LOGIN_SUCCESS,
+    payload: data,
+  };
+};
+export const loginFails = (error) => {
+  return {
+    type: LOGIN_FAIL,
+    payload: error,
+  };
+};
```

Comment:
I think that would cause many imports in the other files which could add a lot of code lines and more unnecessary files which somehow would be confusing to work with.